### PR TITLE
Home Assistant Best practice changes

### PR DESCRIPTION
**Pull request by Bre77**
Just a few best practice changes I would like to suggest. When I get more time I was going to try fill the gaps in your test coverage too.

I'll admit, I think I went a little overboard. If splitting out the classes is too much, my first commit still had some good improvements using attributes.

### CODE REVIEW

File: homeassistant/components/aussie_broadband/sensor.py
Comment (nickw444) [inline homeassistant/components/aussie_broadband/sensor.py:17]:
```
# The state represents a monotonically increasing total, e.g. an amount of consumed gas
STATE_CLASS_TOTAL_INCREASING: Final = "total_increasing"
```

Since this does reset monthly, is it truly monotonic increasing?

File: homeassistant/components/aussie_broadband/sensor.py
Comment (nickw444) [inline homeassistant/components/aussie_broadband/sensor.py:101]: Perhaps `_attr_last_reset` would be appropriate to pair with this? (and for the others)

**Review by nickw444 — APPROVED** [review]
Looks good, will merge after discussing the two questions

File: homeassistant/components/aussie_broadband/sensor.py
Comment (Bre77) [reply homeassistant/components/aussie_broadband/sensor.py:17]: Yeah, so by definition, when a total_increasing entity goes down, its assumed to have reset. This means its possible for Home Assistant to show download usage by day/week/month irrespective of the billing cycling. This was only just introduced into dev recently.

File: homeassistant/components/aussie_broadband/sensor.py
Comment (Bre77) [reply homeassistant/components/aussie_broadband/sensor.py:101]: Not anymore, see https://developers.home-assistant.io/blog/2021/08/16/state_class_total/
last_reset is being removed already.

File: homeassistant/components/aussie_broadband/sensor.py
Comment (nickw444) [reply homeassistant/components/aussie_broadband/sensor.py:17]: Excellent! This is awesome 💯  Thanks for linking to the blog post below!

**Comment by nickw444** [comment]
😆 just noticed you added more test coverage, which I worked on last night but forgot to commit + push 🤦

**Comment by nickw444** [comment]
These changes are looking great! Love that you added the reauth change too, that's something I hadn't considered

**Comment by Bre77** [comment]
@nickw444 I am having issues with the reauth configflow tests, so will keep working through that. Everything else I have worked on has been pushed.

**Comment by Bre77** [comment]
If I can just solve the reauth test, we will be at 100%.

Name                                                       Stmts   Miss  Cover   Missing
----------------------------------------------------------------------------------------
homeassistant/components/aussie_broadband/__init__.py         28      0   100%
homeassistant/components/aussie_broadband/config_flow.py      52      5    90%   100-101, 109-113
homeassistant/components/aussie_broadband/const.py             4      0   100%
homeassistant/components/aussie_broadband/sensor.py           53      0   100%
----------------------------------------------------------------------------------------
TOTAL                                                        137      5    96%

**Comment by Bre77** [comment]
@nickw444 we have reached 100% test coverage. Please merge this when you are ready. If I do any more improvements I will raise them in a new PR so you can get this one into the PR and finalise the review process there.

As I said in the review, I am happy for you to add me as a codeowner, as I will be able to help maintain, test and review this moving forward.

File: homeassistant/components/aussie_broadband/config_flow.py
Comment (nickw444) [inline homeassistant/components/aussie_broadband/config_flow.py:103]: Should we prefix with `_` to indicate a private member?

```suggestion
    async def _create_entry(self, service):
```

File: homeassistant/components/aussie_broadband/config_flow.py
Comment (nickw444) [inline homeassistant/components/aussie_broadband/config_flow.py:115]: Neat, that's handy

File: tests/components/aussie_broadband/test_config_flow.py
Comment (nickw444) [inline tests/components/aussie_broadband/test_config_flow.py:100]: https://github.com/nickw444/home-assistant/blob/d16c1350a5ac748b9a7c7acb709abfd2b6d5ec57/tests/components/aussie_broadband/test_config_flow.py#L156-L178

I added a test for this last night (forgot to push). I'll drop mine in a follow up commit, they are identical except for the assertion about `mock_setup_entry` calls and `assert result2["reason"] == "no_services_found"`

**Comment by nickw444** [comment]
Thanks so much for the contributions, I've learnt a lot!

**Pull request merged**
